globus_automate_client/cli/flows.py

```diff
@@ -170,6 +170,14 @@ def flow_deploy(
         case_sensitive=False,
         show_default=True,
     ),
+    dry_run: bool = typer.Option(
+        False,
+        "--dry-run",
+        help=(
+            "Do a dry run of deploying the flow to test your definition without"
+            " actually making changes."
```

Comment:
I think it would be useful to add more information about what this is validating, beyond the info a user would get just trying to deploy a Flow like normal.

Reply:
@rudyardrichter can you elaborate a little on what `--dry-run` gives feedback on? I still feel like we could be clearer about it in the help text.